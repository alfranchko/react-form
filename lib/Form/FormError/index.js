import _Object$getPrototypeOf from 'babel-runtime/core-js/object/get-prototype-of';
import _classCallCheck from 'babel-runtime/helpers/classCallCheck';
import _createClass from 'babel-runtime/helpers/createClass';
import _possibleConstructorReturn from 'babel-runtime/helpers/possibleConstructorReturn';
import _inherits from 'babel-runtime/helpers/inherits';
import _CProjectsReactFormLayerNode_modulesBabelPresetReactHmreNode_modulesRedboxReactLibIndexJs from 'C:\\Projects\\react-form-layer\\node_modules\\babel-preset-react-hmre\\node_modules\\redbox-react\\lib\\index.js';
import _CProjectsReactFormLayerNode_modulesReactTransformCatchErrorsLibIndexJs from 'C:\\Projects\\react-form-layer\\node_modules\\react-transform-catch-errors\\lib\\index.js';
import _react from 'react';
import _CProjectsReactFormLayerNode_modulesReactTransformHmrLibIndexJs from 'C:\\Projects\\react-form-layer\\node_modules\\react-transform-hmr\\lib\\index.js';

var _class, _temp;

var _components = {
    FormError: {
        displayName: 'FormError'
    }
};

var _CProjectsReactFormLayerNode_modulesReactTransformHmrLibIndexJs2 = _CProjectsReactFormLayerNode_modulesReactTransformHmrLibIndexJs({
    filename: 'C:/Projects/react-form-layer/src/Form/FormError/index.js',
    components: _components,
    locals: [module],
    imports: [_react]
});

var _CProjectsReactFormLayerNode_modulesReactTransformCatchErrorsLibIndexJs2 = _CProjectsReactFormLayerNode_modulesReactTransformCatchErrorsLibIndexJs({
    filename: 'C:/Projects/react-form-layer/src/Form/FormError/index.js',
    components: _components,
    locals: [],
    imports: [_react, _CProjectsReactFormLayerNode_modulesBabelPresetReactHmreNode_modulesRedboxReactLibIndexJs]
});

function _wrapComponent(id) {
    return function (Component) {
        return _CProjectsReactFormLayerNode_modulesReactTransformHmrLibIndexJs2(_CProjectsReactFormLayerNode_modulesReactTransformCatchErrorsLibIndexJs2(Component, id), id);
    };
}

import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

var FormError = _wrapComponent('FormError')((_temp = _class = function (_Component) {
    _inherits(FormError, _Component);

    function FormError() {
        _classCallCheck(this, FormError);

        return _possibleConstructorReturn(this, (FormError.__proto__ || _Object$getPrototypeOf(FormError)).apply(this, arguments));
    }

    _createClass(FormError, [{
        key: 'renderFirstError',
        value: function renderFirstError(errors) {
            var first = void 0;

            for (first in errors) {
                break;
            }return errors[first];
        }
    }, {
        key: 'render',
        value: function render() {
            var _props = this.props,
                className = _props.className,
                children = _props.children;
            var _context$formData = this.context.formData,
                errors = _context$formData.errors,
                hasMistake = _context$formData.hasMistake;

            var _classname = classnames('form--globalError', className);

            return hasMistake ? React.createElement(
                'div',
                { className: _classname },
                children ? children(errors, hasMistake) : this.renderFirstError(errors)
            ) : null;
        }
    }]);

    return FormError;
}(Component), _class.propTypes = {
    className: PropTypes.string,
    // Интересный прием передавать функцию как children чтобы переопределить render. Так же мы делаем в AsyncBundle
    children: PropTypes.func
}, _class.contextTypes = {
    formData: PropTypes.object.isRequired
}, _temp));

export default FormError;