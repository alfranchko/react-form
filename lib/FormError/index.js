'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _class, _temp;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var FormError = (_temp = _class = function (_Component) {
    _inherits(FormError, _Component);

    function FormError() {
        _classCallCheck(this, FormError);

        return _possibleConstructorReturn(this, (FormError.__proto__ || Object.getPrototypeOf(FormError)).apply(this, arguments));
    }

    _createClass(FormError, [{
        key: 'renderFirstError',
        value: function renderFirstError(errors) {
            var first = void 0;

            for (first in errors) {
                break;
            }return errors[first];
        }
    }, {
        key: 'render',
        value: function render() {
            var _props = this.props,
                className = _props.className,
                children = _props.children;
            var _context$formData = this.context.formData,
                errors = _context$formData.errors,
                hasMistake = _context$formData.hasMistake;

            var _classname = (0, _classnames2.default)('form--globalError', className);

            return hasMistake ? _react2.default.createElement(
                'div',
                { className: _classname },
                children ? children(errors, hasMistake) : this.renderFirstError(errors)
            ) : null;
        }
    }]);

    return FormError;
}(_react.Component), _class.propTypes = {
    className: _propTypes2.default.string,
    // Интересный прием передавать функцию как children чтобы переопределить render. Так же мы делаем в AsyncBundle
    children: _propTypes2.default.func
}, _class.contextTypes = {
    formData: _propTypes2.default.object.isRequired
}, _temp);
exports.default = FormError;