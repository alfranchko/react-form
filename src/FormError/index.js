import React, {Component} from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'

class FormError extends Component {

    static propTypes = {
        className: PropTypes.string,
        // Интересный прием передавать функцию как children чтобы переопределить render. Так же мы делаем в AsyncBundle
        children: PropTypes.func
    }

    static contextTypes = {
        formData: PropTypes.object.isRequired
    }

    renderFirstError(errors) {
        let first

        for (first in errors) break

        return errors[first]
    }

    render() {
        const {className, children} = this.props
        const {errors, hasMistake} = this.context.formData
        const _classname = classnames('form--globalError', className)

        return hasMistake ?
            <div className={_classname}>
                {children ? children(errors, hasMistake) : this.renderFirstError(errors)}
            </div>
            : null
    }

}

export default FormError