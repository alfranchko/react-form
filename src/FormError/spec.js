import React from 'react'
import {shallow} from 'enzyme'
import FormError from './index'

describe('блок с ошибками FormError', () => {
    const defaultContext = {
        formData: {
            errors: '',
            hasMistake: false
        }
    }

    const setup = (props, context) => {
        const wrapper = shallow(<FormError {...props} />, {
            context: {
                ...context
            }
        })
        const instance = wrapper.instance()

        return {wrapper, props, instance}
    }

    test('сделать snapshot', () => {
        const {wrapper} = setup(null, defaultContext)
        expect(wrapper).toMatchSnapshot()
    })

    test('если функция не передана и есть ошибка, то вызывать renderFirstError', () => {
        const {wrapper, instance} = setup(null, {
            formData: {
                errors: ['error'],
                hasMistake: true
            }
        })
        jest.spyOn(instance, 'renderFirstError')
        wrapper.setProps()
        expect(instance.renderFirstError.mock.calls.length).toBe(1)
    })

    test('если функция не передана и есть ошибка, то вызывать renderFirstError и рендерить первую ошибку', () => {
        const {wrapper, instance} = setup(null, {
            formData: {
                errors: ['error1', 'error2'],
                hasMistake: true
            }
        })
        expect(wrapper.find('.form--globalError')).toMatchSnapshot()
    })

})