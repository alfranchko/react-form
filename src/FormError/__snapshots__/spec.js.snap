// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`блок с ошибками FormError если функция не передана и есть ошибка, то вызывать renderFirstError и рендерить первую ошибку 1`] = `
<div
  className="form--globalError"
>
  error1
</div>
`;

exports[`блок с ошибками FormError сделать snapshot 1`] = `""`;
